src/utils/actions.js

```diff
@@ -0,0 +1,321 @@
+          return evaluateExpression(expression, resource);
+        });
+      } else {
+        // default true for non-applicability conditions
+        return true;
+      }
+    });
+    // boolMap should be all-true (all conditions met)
+    const triggered = boolMap.every(entry => {
+      return entry === true;
+    });
+    context.flags['triggered'] = triggered;
+  },
+  'check-participant-registration': context => {
+    // TODO: how do we check the participant registration?
+    // Through the source client?
+    return context;
+  },
+  'create-report': context => {
+    // the output might define a profile
+    let contentBundle = createBundle(context.records, 'collection');
+    if (context.action.output) {
+      const output = context.action.output[0];
+      if (output.profile) contentBundle = validateProfile(contentBundle, output.profile[0]);
+    }
+    const header = makeHeader(context);
+    context.contentBundle = contentBundle;
+    context.reportingBundle = createBundle([header, ...contentBundle.entry], 'message');
+  },
+  'validate-report': context => {
+    // check if fhir is well-formed
+    const validation = fhir.validate(context.reportingBundle);
+    context.flags['valid'] = validation.valid;
+  },
+  'submit-report': async context => {
+    await context.client.dest
+      .submit(context.reportingBundle)
+      .then(
+        result => {
+          if (result.status === 200 || result.status === 202) {
+            context.flags['submitted'] = true;
+          }
+        },
+        () => {
+          context.flags['submitted'] = false;
+        }
+      )
+      .catch(() => {
+        context.flags['submitted'] = false;
+      });
+  },
+  'deidentify-report': async context => {
+    const client = context.client.trust;
+    await client
+      .deidentify(context.reportingBundle)
+      .then(
+        result => {
+          context.reportingBundle = result.data;
+          context.flags['deidentified'] = true;
+        },
+        () => {
+          context.flags['deidentified'] = false;
+        }
+      )
+      .catch(() => {
+        context.flags['deidentified'] = false;
+      });
+  },
+  'anonymize-report': context => {
+    const client = context.client.trust;
+    client
+      .anonymize(context.reportingBundle)
+      .then(
+        result => {
+          context.reportingBundle = result.data;
+          context.flags['anonymized'] = true;
+        },
+        () => {
+          context.flags['anonymized'] = false;
+        }
+      )
+      .catch(() => {
+        context.flags['anonymized'] = false;
+      });
+  },
+  'pseudonymize-report': context => {
+    const client = context.client.trust;
+    client
+      .pseudonymize(context.reportingBundle)
+      .then(
+        result => {
+          context.reportingBundle = result.data;
+          context.flags['pseudonymized'] = true;
+        },
+        () => {
+          context.flags['pseudonymize'] = false;
```

Comment:
This key doesn't match up with the one right above it. Same with the pseudonymize a few lines below.